test/spec/spectest.cpp

```diff
@@ -188,25 +192,26 @@ parseValueList(const rapidjson::Value &Args) {
 
 // Helper function to parse parameters from json to vector of string pair.
 std::vector<std::pair<std::string, std::string>>
-parseExpectedList(const rapidjson::Value &Args) {
+parseExpectedList(const simdjson::dom::array &Args) {
   std::vector<std::pair<std::string, std::string>> Result;
-  Result.reserve(Args.Size());
-  for (const auto &Element : Args.GetArray()) {
-    const auto &Type = Element["type"].Get<std::string>();
-    const auto &ValueNode = Element["value"];
-    if (ValueNode.IsString()) {
-      Result.emplace_back(Type, ValueNode.Get<std::string>());
-    } else if (ValueNode.IsArray()) {
-      const auto LaneType = Element["lane_type"].Get<std::string>();
-      std::string Value;
-      for (auto Iter = ValueNode.Begin(); Iter != ValueNode.End(); ++Iter) {
-        Value += Iter->Get<std::string>();
-        Value += ' ';
+  Result.reserve(Args.size());
+  for (const simdjson::dom::object &Element : Args) {
+    std::string_view Type = Element["type"];
+    simdjson::dom::element Value = Element["value"];
+    if (Value.type() == simdjson::dom::element_type::ARRAY) {
+      simdjson::dom::array ValueNodeArray = Value;
+      std::string StrValue;
+      std::string_view LaneType = Element["lane_type"];
+      for (std::string_view X : ValueNodeArray) {
+        StrValue += std::string(X);
+        StrValue += ' ';
       }
-      Value.pop_back();
-      Result.emplace_back(Type + LaneType, std::move(Value));
-    } else {
-      assumingUnreachable();
```

Comment:
You should keep `assumingUnreachable()`.

Reply:
Added missing `assumingReachable()` in [2def666](https://github.com/WasmEdge/WasmEdge/pull/2659/commits/2def666c877c49e25752cbebb63f04a8ad8b45c8)